src/AutoRest.CSharp/Output/Models/Types/OutputLibrary.cs

```diff
@@ -11,125 +11,35 @@
-        private Dictionary<Operation, LongRunningOperation>? _operations;
-        private Dictionary<Operation, ResponseHeaderGroupType>? _headerModels;
 
-        public OutputLibrary(CodeModel codeModel, BuildContext context)
+        protected OutputLibrary (CodeModel codeModel, BuildContext context)
         {
             _codeModel = codeModel;
             _context = context;
         }
 
-        public IEnumerable<TypeProvider> Models => SchemaMap.Values;
-
-        public IEnumerable<RestClient> RestClients => EnsureRestClients().Values;
-
-        public IEnumerable<Client> Clients => EnsureClients().Values;
-
-        public IEnumerable<LongRunningOperation> LongRunningOperations => EnsureLongRunningOperations().Values;
-
-        public IEnumerable<ResponseHeaderGroupType> HeaderModels => (_headerModels ??= EnsureHeaderModels()).Values;
-
-        private Dictionary<Operation, ResponseHeaderGroupType> EnsureHeaderModels()
-        {
-            if (_headerModels != null)
-            {
-                return _headerModels;
-            }
-
-            _headerModels = new Dictionary<Operation, ResponseHeaderGroupType>();
-            foreach (var operationGroup in _codeModel.OperationGroups)
-            {
-                foreach (var operation in operationGroup.Operations)
-                {
-                    var headers = ResponseHeaderGroupType.TryCreate(operationGroup, operation, _context);
-                    if (headers != null)
-                    {
-                        _headerModels.Add(operation, headers);
-                    }
-                }
-            }
-
-            return _headerModels;
-        }
-
-        private Dictionary<Operation, LongRunningOperation> EnsureLongRunningOperations()
-        {
-            if (_operations != null)
-            {
-                return _operations;
-            }
-
-            _operations = new Dictionary<Operation, LongRunningOperation>();
-
-            if (_context.Configuration.PublicClients)
-            {
-                foreach (var operationGroup in _codeModel.OperationGroups)
-                {
-                    foreach (var operation in operationGroup.Operations)
-                    {
-                        if (operation.IsLongRunning)
-                        {
-                            _operations.Add(operation, new LongRunningOperation(operationGroup, operation, _context));
-                        }
-                    }
-                }
-            }
-
-            return _operations;
-        }
-
-        private Dictionary<OperationGroup, Client> EnsureClients()
-        {
-            if (_clients != null)
-            {
-                return _clients;
-            }
+        public virtual IEnumerable<Client> Clients => Enumerable.Empty<Client>();
+        public virtual IEnumerable<LongRunningOperation> LongRunningOperations => Enumerable.Empty<LongRunningOperation>();
```

Comment:
I think this is my largest concern. We would keep adding everything on the base type.